Userland/Services/WindowServer/Window.cpp

```diff
@@ -911,15 +911,11 @@ bool Window::set_untiled()
     VERIFY(!resize_aspect_ratio().has_value());
 
     m_tile_type = WindowTileType::None;
-    set_rect(m_floating_rect);
-
-    Core::EventLoop::current().post_event(*this, make<ResizeEvent>(m_rect));
-    Core::EventLoop::current().post_event(*this, make<MoveEvent>(m_rect));
-
+    tile_type_changed();
     return true;
 }
 
-void Window::set_tiled(WindowTileType tile_type)
+void Window::set_tiled(WindowTileType tile_type, Screen const* cursor_screen)
```

Comment:
`cursor_screen` is use case name; being so specific seems not to be really needed here…? (and below). consider just `screen` (or different specifier if unclear without any).